runners/flink/src/main/java/org/apache/beam/runners/flink/metrics/FlinkMetricContainer.java

```diff
@@ -144,9 +145,9 @@ private void updateGauge(Iterable<MetricResult<GaugeResult>> gauges) {
     }
   }
 
-  private static String getFlinkMetricNameString(String prefix, MetricResult<?> metricResult) {
-    return prefix
-        + METRIC_KEY_SEPARATOR
+  @VisibleForTesting
+  static String getFlinkMetricNameString(MetricResult<?> metricResult) {
+    return METRIC_KEY_SEPARATOR
```

Comment:
With this I get 
```
10.100.208.242.taskmanager.4f13adf64e7315b7198911465ca44119.BeamApp-mwylde-1207183134-fd6cc74c.group.0..group.org.apache.beam.runners.core.ReduceFnRunner.droppedDueToClosedWindow
```

It appears that Flink already includes the step, so in the cases I've seen this can be 

```
    return
        metricResult.getName().getNamespace()
        + METRIC_KEY_SEPARATOR
        + metricResult.getName().getName();
```

which gives you 

```
10.100.208.242.taskmanager.4f13adf64e7315b7198911465ca44119.BeamApp-mwylde-1207185706-ce63a421.group.0.org.apache.beam.runners.core.ReduceFnRunner.droppedDueToClosedWindow: 112
```

Reply:
Flink generates its MetricGroup from the operator name which contains the step id. I think it is fine to omit the step name from the generated name.